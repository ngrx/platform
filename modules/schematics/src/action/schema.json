{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxAction",
  "title": "NgRx Action Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the action.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the action?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files.",
      "default": false
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "group": {
      "type": "boolean",
      "default": false,
      "description": "Group actions file within 'actions' folder",
      "aliases": ["g"]
    },
    "api": {
      "type": "boolean",
      "default": false,
      "description": "Specifies if api success and failure actions should be generated.",
      "aliases": ["a"],
      "x-prompt": "Should we generate success and failure actions?"
    },
    "creators": {
      "type": "boolean",
      "default": true,
      "description": "Specifies whether to use creator functions for handling actions and reducers.",
      "aliases": ["c"],
      "x-prompt": "Do you want to use the create function?"
    }
  },
  "required": []
}
