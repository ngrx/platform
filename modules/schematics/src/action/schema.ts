export interface Schema {
  /**
   * The name of the component.
   */

  name: string;
  /**
   * The path to create the component.
   */

  path?: string;
  /**
   * The name of the project.
   */
  project?: string;
  /**
   * Specifies if a spec file is generated.
   */
  spec?: boolean;
  /**
   * Flag to indicate if a dir is created.
   */

  flat?: boolean;
  /**
   * Group actions file within 'actions' folder
   */

  group?: boolean;
}
