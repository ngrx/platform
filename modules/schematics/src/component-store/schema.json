{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxComponentStore",
  "title": "NgRx Component Store Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the component store.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the component store?"
    },
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component store.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "skipTests": {
      "type": "boolean",
      "default": false,
      "description": "When true, does not create test files."
    },
    "component": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring component.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which component (path) should the component store be provided in?"
    },
    "module": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath",
      "x-prompt": "To which module (path) should the component store be provided in?"
    }
  },
  "required": []
}
