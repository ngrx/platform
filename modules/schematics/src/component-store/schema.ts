export interface Schema {
  /**
   * The name of the component store.
   */
  name: string;

  /**
   * The path to create the component store.
   */
  path?: string;

  /**
   * The name of the project.
   */
  project?: string;

  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;

  /**
   * When true, does not create test files.
   */
  skipTests?: boolean;

  /**
   * Allows specification of the declaring component.
   */
  component?: string;

  /**
   * Allows specification of the declaring module.
   */
  module?: string;
}
