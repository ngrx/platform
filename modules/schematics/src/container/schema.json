{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxContainer",
  "title": "NgRx Container Options Schema",
  "type": "object",
  "properties": {
    "path": {
      "type": "string",
      "format": "path",
      "description": "The path to create the component.",
      "visible": false
    },
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "aliases": ["p"]
    },
    "name": {
      "type": "string",
      "description": "The name of the component.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What should be the name of the container component?"
    },
    "inlineStyle": {
      "description": "Specifies if the style will be in the ts file.",
      "type": "boolean",
      "alias": "s"
    },
    "inlineTemplate": {
      "description": "Specifies if the template will be in the ts file.",
      "type": "boolean",
      "alias": "t"
    },
    "viewEncapsulation": {
      "description": "Specifies the view encapsulation strategy.",
      "enum": ["Emulated", "Native", "None"],
      "type": "string",
      "alias": "v"
    },
    "changeDetection": {
      "description": "Specifies the change detection strategy.",
      "enum": ["Default", "OnPush"],
      "type": "string",
      "alias": "c"
    },
    "prefix": {
      "type": "string",
      "format": "html-selector",
      "description": "The prefix to apply to generated selectors.",
      "alias": "p"
    },
    "style": {
      "description": "The file extension or preprocessor to use for style files.",
      "type": "string"
    },
    "skipTests": {
      "type": "boolean",
      "description": "When true, does not create test files."
    },
    "flat": {
      "type": "boolean",
      "description": "Flag to indicate if a dir is created."
    },
    "skipImport": {
      "type": "boolean",
      "description": "Flag to skip the module import."
    },
    "selector": {
      "type": "string",
      "format": "html-selector",
      "description": "The selector to use for the component."
    },
    "module": {
      "type": "string",
      "description": "Allows specification of the declaring module.",
      "alias": "m"
    },
    "export": {
      "type": "boolean",
      "description": "Specifies if declaring module exports the component."
    },
    "state": {
      "type": "string",
      "description": "Specifies the path to the state exports."
    },
    "stateInterface": {
      "type": "string",
      "default": "State",
      "description": "Specifies the interface for the state."
    },
    "testDepth": {
      "description": "Specifies whether to create a unit test or an integration test.",
      "enum": ["unit", "integration"],
      "type": "string",
      "default": "integration"
    }
  },
  "required": []
}
