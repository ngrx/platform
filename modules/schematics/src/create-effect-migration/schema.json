{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxCreateEffectMigration",
  "title": "NgRx Effect Migration from @Effect to createEffect",
  "type": "object",
  "properties": {},
  "required": []
}
