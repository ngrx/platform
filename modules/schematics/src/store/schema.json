{
  "$schema": "http://json-schema.org/schema",
  "id": "SchematicsNgRxState",
  "title": "NgRx State Management Options Schema",
  "type": "object",
  "properties": {
    "name": {
      "type": "string"
    },
    "path": {
      "type": "string",
      "default": "app"
    },
    "appRoot": {
      "type": "string"
    },
    "sourceDir": {
      "type": "string",
      "default": "src"
    },
    "flat": {
      "type": "boolean",
      "default": true,
      "description": "Flag to indicate if a dir is created."
    },
    "spec": {
      "type": "boolean",
      "default": true,
      "description": "Specifies if a spec file is generated."
    },
    "module": {
      "type": "string",
      "default": "",
      "description": "Allows specification of the declaring module.",
      "alias": "m",
      "subtype": "filepath"
    },
    "statePath": {
      "type": "string",
      "default": "reducers"
    },
    "root": {
      "type": "boolean",
      "default": false,
      "description": "Flag to setup the root state or feature state."
    }
  },
  "required": [
    "name"
  ]
}
