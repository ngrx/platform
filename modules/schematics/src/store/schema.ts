export interface Schema {
  /**
   * The name of the component.
   */

  name: string;
  /**
   * The path to create the effect.
   */

  path?: string;
  /**
   * The name of the project.
   */
  project?: string;
  /**
   * Flag to indicate if a dir is created.
   */
  flat?: boolean;
  /**
   * Specifies if a spec file is generated.
   */
  spec?: boolean;
  /**
   * Allows specification of the declaring module.
   */
  module?: string;
  /**
   * Specifies the dir for the state folder
   */

  statePath?: string;
  /**
   * Specifies whether this is the root state or feature state
   */

  root?: boolean;
  /**
   * Specifies the interface for the state
   */
  stateInterface?: string;
}
