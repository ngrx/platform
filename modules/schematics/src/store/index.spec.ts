import { Tree, VirtualTree } from '@angular-devkit/schematics';
import { SchematicTestRunner } from '@angular-devkit/schematics/testing';
import * as path from 'path';
import { createAppModule, getFileContent } from '../utility/test';
import { Schema as StoreOptions } from './schema';

describe('Store Schematic', () => {
  const schematicRunner = new SchematicTestRunner(
    '@ngrx/schematics',
    path.join(__dirname, '../../collection.json')
  );
  const defaultOptions: StoreOptions = {
    name: 'foo',
    path: 'app',
    sourceDir: 'src',
    spec: true,
    module: undefined,
    flat: false,
    root: true,
  };

  let appTree: Tree;

  beforeEach(() => {
    appTree = new VirtualTree();
    appTree = createAppModule(appTree);
  });

  it('should create the initial store setup', () => {
    const options = { ...defaultOptions };

    const tree = schematicRunner.runSchematic('store', options, appTree);
    const files = tree.files;
    expect(files.indexOf('/src/app/reducers/index.ts')).toBeGreaterThanOrEqual(
      0
    );
  });

  it('should not be provided by default', () => {
    const options = { ...defaultOptions };

    const tree = schematicRunner.runSchematic('store', options, appTree);
    const content = getFileContent(tree, '/src/app/app.module.ts');
    expect(content).not.toMatch(
      /import { reducers, metaReducers } from '\.\/reducers';/
    );
  });

  it('should import into a specified module', () => {
    const options = { ...defaultOptions, module: 'app.module.ts' };

    const tree = schematicRunner.runSchematic('store', options, appTree);
    const content = getFileContent(tree, '/src/app/app.module.ts');
    expect(content).toMatch(
      /import { reducers, metaReducers } from '\.\/reducers';/
    );
  });

  it('should fail if specified module does not exist', () => {
    const options = { ...defaultOptions, module: '/src/app/app.moduleXXX.ts' };
    let thrownError: Error | null = null;
    try {
      schematicRunner.runSchematic('store', options, appTree);
    } catch (err) {
      thrownError = err;
    }
    expect(thrownError).toBeDefined();
  });

  it('should import a feature a specified module', () => {
    const options = { ...defaultOptions, root: false, module: 'app.module.ts' };

    const tree = schematicRunner.runSchematic('store', options, appTree);
    const content = getFileContent(tree, '/src/app/app.module.ts');
    expect(content).toMatch(
      /StoreModule\.forFeature\('foo', fromFoo\.reducers, { metaReducers: fromFoo.metaReducers }\)/
    );
  });

  it('should use a wildcard for a feature import ', () => {
    const options = { ...defaultOptions, root: false, module: 'app.module.ts' };

    const tree = schematicRunner.runSchematic('store', options, appTree);
    const content = getFileContent(tree, '/src/app/app.module.ts');
    expect(content).toMatch(/import \* as fromFoo from '\.\/reducers';/);
  });
});
