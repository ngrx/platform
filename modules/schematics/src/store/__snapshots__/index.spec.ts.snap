// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store Schematic should add a feature key if not root 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should add the initial config correctly into an empty module 1`] = `
"
        import { NgModule, isDevMode } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { reducers, metaReducers } from './reducers';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

        @NgModule({
          declarations: [],
          imports: [StoreModule.forRoot(reducers, { metaReducers }), isDevMode() ? StoreDevtoolsModule.instrument() : []],
        })
        export class EmptyModule {}
      "
`;

exports[`Store Schematic should create the initial store setup 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should create the initial store setup 2`] = `
"import { isDevMode } from '@angular/core';
import {
  ActionReducer,
  ActionReducerMap,
  createFeatureSelector,
  createSelector,
  MetaReducer
} from '@ngrx/store';


export interface State {

}

export const reducers: ActionReducerMap<State> = {

};


export const metaReducers: MetaReducer<State>[] = isDevMode() ? [] : [];
"
`;

exports[`Store Schematic should import a feature a specified module 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import * as fromFoo from './reducers';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forFeature(fromFoo.fooFeatureKey, fromFoo.reducers, { metaReducers: fromFoo.metaReducers })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should import into a specified module 1`] = `
"import { NgModule, isDevMode } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import { reducers, metaReducers } from './reducers';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forRoot(reducers, { metaReducers }),
    isDevMode() ? StoreDevtoolsModule.instrument() : []
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should not add a feature key if root 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should not skip the initial store setup files if the minimal flag is provided with a feature 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import * as fromFoo from './reducers';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forFeature(fromFoo.fooFeatureKey, fromFoo.reducers, { metaReducers: fromFoo.metaReducers })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should skip the initial store setup files if the minimal flag is provided 1`] = `
"import { NgModule, isDevMode } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { StoreModule } from '@ngrx/store';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    StoreModule.forRoot({}),
    isDevMode() ? StoreDevtoolsModule.instrument() : []
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should support a custom feature state interface name 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should support a custom root state interface name 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should support a default feature state interface name 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;

exports[`Store Schematic should support a default root state interface name 1`] = `
"import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
"
`;
